File: frontend/src/types/analysis.ts
export type SpeechEventType = "repetition" | "prolongation";

export type SpeechEvent = {
  id: string;
  type: SpeechEventType;
  start: number;
  end: number;
  confidence: number;
  text: string;
};

export type AnalysisMetrics = {
  total_events: number;
  repetitions: number;
  prolongations: number;
  speech_rate: number;
  pause_ratio: number;
};

export type AnalysisFeedback = {
  summary: string;
  strengths: string[];
  observations: string[];
  next_step: string;
};

export type AnalysisResult = {
  analysis_id: string;
  exercise_id: string;
  duration_seconds: number;
  transcript: string;
  metrics: AnalysisMetrics;
  events: SpeechEvent[];
  patterns: string[];
  feedback: AnalysisFeedback;
};

export type HealthResponse = {
  status: string;
};

File: backend/app/schemas/analysis.py
from enum import Enum

from pydantic import BaseModel, Field


class SpeechEventType(str, Enum):
    repetition = "repetition"
    prolongation = "prolongation"


class SpeechEvent(BaseModel):
    id: str
    type: SpeechEventType
    start: float
    end: float
    confidence: float
    text: str


class AnalysisMetrics(BaseModel):
    total_events: int
    repetitions: int
    prolongations: int
    speech_rate: float
    pause_ratio: float


class AnalysisFeedback(BaseModel):
    summary: str
    strengths: list[str]
    observations: list[str]
    next_step: str


class AnalysisResult(BaseModel):
    analysis_id: str
    exercise_id: str
    duration_seconds: float
    transcript: str
    metrics: AnalysisMetrics
    events: list[SpeechEvent]
    patterns: list[str] = Field(default_factory=list)
    feedback: AnalysisFeedback
